/* Protocol between libxsocket and Xcache */

message xcache_cmd {
	enum req_type {
		XCACHE_STORE = 1;
		XCACHE_NEWSLICE = 2;
		XCACHE_GETCHUNK = 3;
		XCACHE_REQUESTCHUNK = 4;
		XCACHE_TIMEOUT = 5;
		XCACHE_CLEAR = 6;
		XCACHE_RESPONSE = 7;
		XCACHE_ERROR = 8;
		XCACHE_GET_STATUS = 9;
	};

	enum status_type {
		XCACHE_OK = 1;
		XCACHE_ERR_EXISTS = 2;
	};
  
	required req_type cmd = 1;

	optional int32 context_id = 2;
	optional bytes cid = 3;
	optional bytes dag = 4;
	optional bytes data = 5;
	optional int32 ttl = 6;
	optional int32 cache_size = 7;
	optional int32 cache_policy = 8;
	optional status_type status = 9;
};
